packages/byods/src/data-source-client/index.ts

```diff
@@ -86,4 +89,151 @@ export default class DataSourceClient {
   public async delete(id: string): Promise<ApiResponse<void>> {
     return this.httpClient.delete<void>(`${DATASOURCE_ENDPOINT}/${id}`);
   }
+
+  /**
+   * This method refreshes the DataSource token using dataSourceId and tokenLifetimeMinute
+   * @param {string} dataSourceId The id of the data source.
+   * @param {number} tokenLifetimeMinutes The refresh interval in minutes for the data source. Defaults to 60 but less than < 1440
+   * @param {string} nonceGenerator This method accepts an optional nonceGenerator, developer can provide their own nonceGenerator, defaults to randomUUID.
+   * @returns {Promise<void>}
+   * @example
+   * try {
+    const dataSourceId = '123'; // Replace with your actual dataSourceId
+    const tokenLifetimeMinutes = 60; // Optional, replace with your actual dataSourceToken lifetime minutes
+    const result = await dataSourceClient.scheduleJWSTokenRefresh(123, 60, '550e8400-e29b-41d4-a716-446655440000');
+    // Use the cancel function if needed
+    result.cancel();
+  } catch (error) {
+    console.error('Error refreshing data source token:', error);
+  }
+   */
+
+  public async scheduleJWSTokenRefresh(
+    dataSourceId: string,
+    tokenLifetimeMinutes = 60,
+    nonceGenerator: () => string = crypto.randomUUID
```

Comment:
Using crypto.randomUUID instead of uuid. By default we need to provide crypto.randomUUID